src/environs/__init__.py

```diff
@@ -27,7 +41,38 @@
 Subcast = typing.Union[typing.Type, typing.Callable[..., _T], ma.fields.Field]
 FieldType = typing.Type[ma.fields.Field]
 FieldOrFactory = typing.Union[FieldType, FieldFactory]
-ParserMethod = typing.Callable
+
+
+_int = int
+_bool = bool
+_str = str
+_float = float
+_list = typing.List[typing.Any]
+_dict = typing.Dict[str, typing.Any]
+
+
+class ParserMethod(typing.Generic[_T]):
```

Comment:
non-blocking nit: maybe should make this 'private' if it's not meant to be used

```suggestion
class _ParserMethod(typing.Generic[_T]):
```

Reply:
I don't think it's possible without `@overload` which has already been suggested before.